Lib/gftools/builder/__init__.py

```diff
@@ -0,0 +1,341 @@
+
+    sources:
+      - Texturina.glyphs
+      - Texturina-Italic.glyphs
+    axisOrder:
+      - opsz
+      - wght
+    outputDir: "../fonts"
+    familyName: Texturina
+    stat:
+      - name: Width
+        tag: wdth
+        values:
+        - name: UltraCondensed
+          value: 50
+          ...
+      - name: Weight
+        tag: wght
+        values:
+        - name: Hairline
+          rangeMinValue: 1
+          nominalValue: 1
+          rangeMaxValue: 50
+        ...
+
+To build a font family from the command line, use:
+
+    gftools builder path/to/config.yaml
+
+The config file may contain the following keys. The ``sources`` key is
+required, all others have sensible defaults:
+
+* ``sources``: Required. An array of Glyphs, UFO or designspace source files.
+* ``logLevel``: Debugging log level. Defaults to ``INFO``.
+* ``stylespaceFile``: A statmake ``.stylespace`` file. Defaults to none (Use ``gftools.stat``)
+* ``stat``: A STAT table configuration. This may be either a list of axes and
+    values as demonstrated above, or a dictionary mapping each source file to a
+    per-source list.
+* ``buildVariable``: Build variable fonts. Defaults to true.
+* ``buildStatic``: Build static fonts. Defaults to true.
+* ``buildWebfont``: Build WOFF2 fonts. Defaults to ``$buildStatic``.
+* ``outputDir``: Where to put the fonts. Defaults to ``../fonts/``
+* ``vfDir``: Where to put variable fonts. Defaults to ``$outputDir/variable``.
+* ``ttDir``: Where to put TrueType static fonts. Defaults to ``$outputDir/ttf``.
+* ``otDir``: Where to put CFF static fonts. Defaults to ``$outputDir/otf``.
+* ``woffDir``: Where to put WOFF2 static fonts. Defaults to ``$outputDir/webfonts``.
+* ``autohintTTF`: Whether or not to autohint TTF files. Defaults to ``true``.
+* ``axisOrder``: STAT table axis order. Defaults to fvar order.
+* ``familyName``: Family name for variable fonts. Defaults to family name of first source file.
+
+"""
+
+from fontTools.ttLib import TTFont
+from fontmake.font_project import FontProject
+from ufo2ft import CFFOptimization
+from gftools.fix import fix_font
+from gftools.stat import gen_stat_tables
+from fontTools.otlLib.builder import buildStatTable
+import statmake.classes
+import statmake.lib
+from babelfont import Babelfont
+import sys
+import os
+import shutil
+import glyphsLib
+import tempfile
+from ttfautohint.options import parse_args as ttfautohint_parse_args
+from ttfautohint import ttfautohint
+from fontTools.ttLib.woff2 import main as woff2_main
+import logging
+import yaml
+
+
+class GFBuilder:
+    def __init__(self, configfile=None, config=None):
+        if configfile:
+            self.config = yaml.load(open(configfile), Loader=yaml.SafeLoader)
+            if os.path.dirname(configfile):
+                os.chdir(os.path.dirname(configfile))
+        else:
+            self.config = config
+        self.masters = {}
+        self.logger = logging.getLogger("GFBuilder")
+        self.fill_config_defaults()
```

Comment:
Yeah, I'm not sure that would be worthwhile; for one thing, the `fill_config_defaults` still needs to happen since some config parameters are computed from other ones, which you can't just do with keyword args. Moving yaml out does make some sense. But meh. It's fine.